### PR TITLE
Fix string stream crashing when created on nimscript due to last fix

### LINKED ISSUE #19716: StringStream in static context (arc) now broken after recent write fix
Recent fix meaning #19708

It now seemingly attempts to use prepareMutation in the VM (when using arc/orc)  which gives the error `Error: opcCastIntToPtr: regs[rb].kind: rkNodeAddr`,   from here https://github.com/nim-lang/Nim/blob/devel/lib/system/strs_v2.nim#L177

This code was valid before and is still valid when using refc.

### Example:
```nim
import streams
static:
  let s = newStringStream("a")
  echo s.data # only added for illustrative purposes, breaks before this point
```

### PR DESCRIPTION
**Pull request by beef331**
Closes: https://github.com/nim-lang/Nim/issues/19716
Hopefully this is done properly now.

### CODE REVIEW

File: lib/pure/streams.nim [hunk 1]
```diff
@@ -1274,7 +1274,7 @@ else: # after 1.3 or JS not defined
 
     new(result)
     result.data = s
-    when defined(gcOrc) or defined(gcArc):
+    when not defined(js) and not defined(nimscript) and not defined(nimSeqsV2):
```
Comment (Araq) [inline lib/pure/streams.nim]: `not defined(nimSeqsV2)`? Why not

```suggestion
    when (defined(gcOrc) or defined(gcArc)) and not defined(js) and not defined(nimscript):
```

Comment (beef331) [reply lib/pure/streams.nim]: It's what `preparemutations` is [guarded with](https://github.com/nim-lang/Nim/blob/version-1-6/lib/system.nim#L3141).
So figured that it was the best.

Comment (beef331) [reply lib/pure/streams.nim]: In hindsight `when declared(prepareMutation)` seems more 'proper'

**Comment by PMunch** [comment]
@beef331 it doesn't appear like that test is being run with `--gc:arc`. Trying to run that snippet with `--gc:arc` in a separate file with the latest devel still crashes.

**Comment by SirOlaf** [comment]
For my specific test case only this works, but I don't know how complete that is
```nim
when nimvm:
  discard
else:
  prepareMutation(result.data)
```

**Comment by ghost** [comment]
@SirOlaf can you please make a PR? The underlying issue is still not fixed. This works for me:
```nim
    when nimvm:
      discard
    else:
      when declared(prepareMutation):
        prepareMutation(result.data) # Allows us to mutate using `addr` logic like `copyMem`, otherwise it errors.
```

**Pull request merged**